drivers/mdio/mdio_xilinx_axi_ethernet_lite.c

```diff
@@ -0,0 +1,233 @@
+/*
+ * Xilinx AXI Ethernet Lite MDIO
+ *
+ * Copyright(c) 2025, CISPA Helmholtz Center for Information Security
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(eth_xilinx_axi_ethernet_lite_mdio, CONFIG_ETHERNET_LOG_LEVEL);
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/mdio.h>
+#include <zephyr/device.h>
+
+#include <stdint.h>
+
+#define MDIO_XILINX_AXI_ETHERNET_LITE_MAX_PHY_DEVICES 32
+
+#define MDIO_XILINX_AXI_ETHERNET_LITE_MDIO_ADDRESS_REG_OFFSET    0x07e4
+#define MDIO_XILINX_AXI_ETHERNET_LITE_MDIO_WRITE_DATA_REG_OFFSET 0x07e8
+#define MDIO_XILINX_AXI_ETHERNET_LITE_MDIO_READ_DATA_REG_OFFSET  0x07ec
+#define MDIO_XILINX_AXI_ETHERNET_LITE_MDIO_CONTROL_REG_OFFSET    0x07f0
+
+#define MDIO_XILINX_AXI_ETHERNET_LITE_MDIO_CONTROL_REG_MDIO_ENABLE_MASK  BIT(3)
+#define MDIO_XILINX_AXI_ETHERNET_LITE_MDIO_CONTROL_REG_MDIO_BUSY_MASK    BIT(0)
+#define MDIO_XILINX_AXI_ETHERNET_LITE_MDIO_CONTROL_REG_MDIO_DISABLE_MASK 0
+
+#define MDIO_XILINX_AXI_ETHERNET_LITE_MDIO_ADDRESS_REG_OP_READ       BIT(10)
+#define MDIO_XILINX_AXI_ETHERNET_LITE_MDIO_ADDRESS_REG_OP_WRITE      0
+#define MDIO_XILINX_AXI_ETHERNET_LITE_MDIO_ADDRESS_REG_SHIFT_REGADDR 0
+#define MDIO_XILINX_AXI_ETHERNET_LITE_MDIO_ADDRESS_REG_SHIFT_PHYADDR 5
+
+struct mdio_xilinx_axi_ethernet_lite_data {
+	struct k_mutex mutex;
+	bool bus_enabled;
+};
+
+struct mdio_xilinx_axi_ethernet_lite_config {
+	void *reg;
+};
+
+static inline uint32_t
+mdio_xilinx_axi_ethernet_lite_read_reg(const struct mdio_xilinx_axi_ethernet_lite_config *config,
```

Comment:
please shorten these names and also the defines as these are not global or exposed to other files, they don't have to be that unique